### PR TITLE
Refine Flatpak permissions.

### PR DESCRIPTION
**Pull request by Lokawn**
- Refine Flatpak permissions, currently muffon flatpak asks for Read/write home directory access, which I think is not needed. Also permission to talk to system and session bus for system tray icon.
- Use correct category, according to [specification](https://specifications.freedesktop.org/menu-spec/latest/apa.html#main-category-registry).

### CODE REVIEW

**Review by gamersi — DISMISSED** [review]
LGTM, Thanks!
Waiting for @xyloflake and @staniel359 s approval

**Comment by xyloflake** [comment]
Read/Write is needed @Lokawn 

@gamersi I think we both know that electron store uses the file system and needs filesystem access to the hone directory to work currently.

I would suggest one of the **team members** to actually try it out on a flatpak machine. Overall, I think core team should be commiting this instead of outside contributors for security purposes. 

This PR is a no go for me as of now. However if @staniel359 insists, I'll approve.

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:49]: Change com.electron.muffon to com.staniel359.muffon

This is change is part of upcoming v2.0.4

File: electron-builder.json
Comment (xyloflake) [inline electron-builder.json:21]: @Lokawn @gamersi why is AudioVideo needed? We don't need video.

**Review by xyloflake — DISMISSED** [review]

File: electron-builder.json
Comment (staniel359) [reply electron-builder.json:49]: @xyloflake That's for next major version.

File: electron-builder.json
Comment (staniel359) [reply electron-builder.json:21]: @xyloflake But we have video player.

**Comment by staniel359** [comment]
@xyloflake Also I don't insist, just whatever is better for Flatpak community.

File: electron-builder.json
Comment (xyloflake) [reply electron-builder.json:21]: @staniel359
> Application for presenting, creating, or processing multimedia (audio/video)

We do not process the video, the video is processed by the youtube embedder which has some very different method of video processing and it utilizes the chromiul builtins. Which is why video must not be enabled.

File: electron-builder.json
Comment (xyloflake) [reply electron-builder.json:49]: @staniel359 I think we should do it from this version.

Comment (xyloflake) [reply electron-builder.json:49]: Otherwise autoupdates will be messed up by v3. Will explain later.

Comment (staniel359) [reply electron-builder.json:49]: @xyloflake Let's discuss it on Discord.

File: electron-builder.json
Comment (staniel359) [reply electron-builder.json:21]: @xyloflake On Discord.

**Comment by gamersi** [comment]
Yeah, couldn't test it because of the bug mentioned in discord. I can test it when that is fixed

File: electron-builder.json
Comment (Lokawn) [reply electron-builder.json:21]: Notes for Audio category in [specifications](https://specifications.freedesktop.org/menu-spec/latest/apa.html#main-category-registry) states that

> Desktop entry must include AudioVideo as well

Otherwise it does not go to `Multimedia` tab on KDE, but to the `Lost & Found` tab.

**Comment by Lokawn** [comment]
> Read/Write is needed @Lokawn

The flatpak provided on official website does not have access to system files, only to the `/home` directory as mentioned in the [electron builder documentation](https://www.electron.build/configuration/flatpak.html) in finishArgs section.

Flatpak [Documentation](https://github.com/flatpak/flatpak/wiki/Filesystem) states that
> In addition to /app and /usr, each application sandbox gets a per-app data directory mounted which persists over app upgrades. It is stored as a subdirectory of the $HOME/.var/app directory (e.g. $HOME/.var/app/org.gnome.GEdit/). The $XDG_DATA_HOME, $XDG_CONFIG_HOME and $XDG_CACHE_HOME environment variables are set to point at data/, config/ and cache/ subdirectories respectively in this location.

So a make shift Home directory is provided to every app.

If access to any other particular file is needed then it can be whitelisted using `--filesystem=` permission.
`--persist=.` can be used to check for files created by the app.

**Review by staniel359 — DISMISSED** [review]

**Comment by xyloflake** [comment]
As discussed on discord and as instructed by @staniel359, we've decided to keep this open for a while until we get to test it. These reviews were made without proper testing and thus they were dismissed.

@Lokawn thank you for your valuable contribution, we'll merge it ASAP but we need to test it before we merge (will take some good amount of time since we, the devs are busy as of now).

**Comment by xyloflake** [comment]
Coming back to this. Ready for testing. @gamersi I'll let you take the lead for testing this.

**Comment by xyloflake** [comment]
@staniel359 the branch has been rebased.

@Lokawn we'll probably merge this very soon.

**Review by gamersi — CHANGES_REQUESTED** [review]
Hello @Lokawn,

We have tested the application on two machines. It worked fine with Arch Linux + KDE Plasma (Wayland). However, when testing with Arch Linux + GNOME (46.2, Wayland 1.23.0), the application does not perform as expected. The application boundaries are significantly misaligned, and the cursor does not adapt properly. Interestingly, this issue is not present in the main branch. Logs and videos have been attached for your reference.

Logs:
```
(node:3) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
(Use `muffon --trace-warnings ...` to show where the warning was created)
LaunchProcess: failed to execvp:
xdg-settings
APPIMAGE env is not defined, current application is not an AppImage
checkForUpdatesAndNotify called, downloadPromise is null
^C
[3 preload-host-spawn-strategy] Dropping 0x17e400240d80 (3) because of connection closed
[3:0728/205219.516729:FATAL:bus.cc(1246)] D-Bus connection was disconnected. Aborting.
```
Video: [muffon-flatpak-issue.webm](https://files.gamersi.at/muffon-flatpak-issue.webm)

**Comment by gamersi** [comment]
Here is the video:  
[muffon-flatpak-issue.webm](https://files.gamersi.at/muffon-flatpak-issue.webm)

**Review by xyloflake — CHANGES_REQUESTED** [review]
As has been reported by @gamersi, a very weird behaviour related to window scaling has been detected on flatpak packages ONLY. AppImages and other formats worked fine during the test from **the same build**.

We have also checked .flatpak packages from main, but there was no issue.

This directly implies that the problem arises from this PR, that too specifically for flatpak.

**Comment by xyloflake** [comment]
@Lokawn please fix your PR within 3 days or we'll have to close it due to inactivity.

**Comment by xyloflake** [comment]
@Lokawn this PR will be closed due to inactivity.

**Pull request closed without merging**
